driver-core/src/main/com/mongodb/AutoEncryptionSettings.java

```diff
@@ -73,6 +75,14 @@ public final class AutoEncryptionSettings {
     private final boolean bypassAutoEncryption;
     private final Map<String, BsonDocument> encryptedFieldsMap;
     private final boolean bypassQueryAnalysis;
+    /**
+     * Proxy setting used for connecting to Key Management Service via a SOCKS5 proxy server.
+     * <p>
+     * NOTE: If this field is left unset, the default behavior is to utilize the {@link SocketSettings#getProxySettings()}
+     * from the {@link MongoClientSettings} where the {@link AutoEncryptionSettings} are specified.
+     */
+    @Nullable
+    private final ProxySettings proxySettings;
```

Comment:
The [spec](https://github.com/mongodb/specifications/blob/b8a5a00f0fd9c066e5d79eccbed0dfc538796856/source/socks5-support/socks5.rst#connection-establishment) doesn't seem to require this override.  Why did you add it? (It can be added later if requested)